deps/v8/src/base/macros.h

```diff
@@ -173,14 +173,17 @@ namespace base {
 // base::is_trivially_copyable will differ for these cases.
 template <typename T>
 struct is_trivially_copyable {
-#if V8_CC_MSVC
+#if V8_CC_MSVC || (__GNUC__ == 12 && __GNUC_MINOR__ == 2)
   // Unfortunately, MSVC 2015 is broken in that std::is_trivially_copyable can
   // be false even though it should be true according to the standard.
   // (status at 2018-02-26, observed on the msvc waterfall bot).
   // Interestingly, the lower-level primitives used below are working as
   // intended, so we reimplement this according to the standard.
   // See also https://developercommunity.visualstudio.com/content/problem/
   //          170883/msvc-type-traits-stdis-trivial-is-bugged.html.
+  //
+  // GCC 12.2 is broken too, so the same polyfill is used.
+  // See: https://github.com/nodejs/node/pull/45427
```

Comment:
Is this the expected link? It is not obvious to me because it is a long thread. Might be better to link to https://github.com/nodejs/node/issues/53633 instead.

Reply:
It's unfortunate that GitHub folds the comments, but that is thread contains the whole background about:

1. On which GCC version is this broken
2. A minimal reproduction of the broken implementation

https://github.com/nodejs/node/issues/53633 doesn't really reference any of those.

Reply:
Although, @jdapena bisected the GCC fix to https://gcc.gnu.org/git/?p=gcc.git;a=commitdiff;h=aeba3e009b0abfccaf01797556445dbf891cc8dc;hp=2773a90c0f4f323e8ace0593893bc8fcbd2266cf, so that might be a better reference than a Node.js PR anyway.

It also appears that this is broken on 12.1 (at least, it also fails when I tried to use the 12.1 gcc docker container on Debian). However according to @richardlau  12.1 works on RHEL8, maybe they have backported this patch to toolset-gcc-12 on RHEL (pretty likely considering that patch was submitted by someone with a redhat.com email)?